mods/start-button-actions.wh.cpp

```diff
@@ -0,0 +1,1302 @@
+    // Block Win key (MOD_WIN + vk=0) and Ctrl+Esc if in custom mode
+    if (g_windowsKeyAction == "custom") {
+        if ((fsModifiers == MOD_WIN && vk == 0) || 
+            (fsModifiers == MOD_CONTROL && vk == VK_ESCAPE)) {
+            Wh_Log(L"Windows Key Actions: Blocked RegisterHotKey (mod=%u, vk=%u)", fsModifiers, vk);
+            return FALSE; // Pretend it failed
+        }
+    }
+    
+    // Allow other hotkeys through
+    return g_pOriginalRegisterHotKey(hWnd, id, fsModifiers, vk);
+}
+
+// Forward declarations
+LRESULT CALLBACK LowLevelKeyboardProc(int nCode, WPARAM wParam, LPARAM lParam);
+LRESULT CALLBACK LowLevelMouseProc(int nCode, WPARAM wParam, LPARAM lParam);
+void ExecuteCustomAction();
+void ExecuteStartButtonAction();
+void LoadSettings();
+bool IsWindowsKey(DWORD vkCode);
+void ExecuteCommand(const std::string& command);
+HMODULE GetThisModule();
+std::wstring GetModuleDirectory();
+std::string WideToUtf8(const std::wstring& w);
+std::wstring Utf8ToWide(const std::string& s);
+DWORD WINAPI WorkerThreadProc(LPVOID);
+DWORD WINAPI HookThreadProc(LPVOID);
+void CloseStartMenuIfOpen();
+void SendVirtualKey(WORD vk, bool down);
+void ClickStartButton();
+void UpdateStartButtonInfo();
+void DisableShellHotkeys();
+void RestoreShellHotkeys();
+bool ShellExecuteUnelevated(const wchar_t* file, const wchar_t* params, int nShow = SW_SHOWNORMAL);
+bool CreateProcessWithExplorerToken(const wchar_t* app, wchar_t* cmdMutable, int nShow = SW_SHOWNORMAL);
+
+// APC callback to unregister hotkeys from the correct thread context
+void NTAPI UnregisterHotkeysAPC(ULONG_PTR Parameter) {
+    UnregisterHotKey(NULL, 1);  // Win key
+    UnregisterHotKey(NULL, 2);  // Ctrl+Esc
+    Wh_Log(L"Windows Key Actions: Unregistered hotkeys via APC");
+}
+
+// Disable Windows shell hotkeys (Win key and Ctrl+Esc)
+void DisableShellHotkeys() {
+    if (g_hotkeysDisabled.load() || g_windowsKeyAction != "custom") {
+        return;
+    }
+    
+    // Find the window that registered the Start menu hotkeys
+    // On Windows 10/11, this is typically the ApplicationManager window
+    HWND startMenuWnd = FindWindowW(L"Windows.UI.Core.CoreWindow", L"Start");
+    if (!startMenuWnd) {
+        // Try alternate window class for Windows 11
+        startMenuWnd = FindWindowW(L"ApplicationFrameWindow", nullptr);
+    }
+    if (!startMenuWnd) {
+        // Fallback to Immersive Shell window
+        startMenuWnd = FindWindowW(L"ApplicationManager_ImmersiveShellWindow", nullptr);
+    }
+    
+    if (startMenuWnd) {
+        // Get the thread that owns the Start menu window
+        DWORD threadId = GetWindowThreadProcessId(startMenuWnd, nullptr);
+        if (threadId) {
+            HANDLE thread = OpenThread(THREAD_SET_CONTEXT, FALSE, threadId);
+            if (thread) {
+                // Queue an APC to unregister hotkeys in the correct thread context
+                QueueUserAPC(UnregisterHotkeysAPC, thread, 0);
```

Comment:
This doesn't look right, the start menu isn't part of explorer.exe, and the mod only targets explorer.exe. How does this work?